js/ui/number_box/number_box.mask.js

```diff
@@ -462,8 +470,23 @@ var NumberBoxMask = NumberBoxBase.inherit({
 
         eventsEngine.on($input, eventUtils.addNamespace(INPUT_EVENT, NUMBER_FORMATTER_NAMESPACE), this._formatValue.bind(this));
         eventsEngine.on($input, eventUtils.addNamespace("dxclick", NUMBER_FORMATTER_NAMESPACE), function() {
-            this._caret(maskCaret.getCaretInBoundaries(this._caret(), this._getInputVal(), this._getFormatPattern()));
+            var that = this;
+
+            if(!this._caretTimeout) {
+                this._caretTimeout = setTimeout(function() {
+                    that._caret(maskCaret.getCaretInBoundaries(that._caret(), that._getInputVal(), that._getFormatPattern()));
+                }, CARET_TIMEOUT_DURATION);
```

Comment:
function() {).bind(this)
```suggestion
                function() {
                  this._caret(maskCaret.getCaretInBoundaries(this._caret(), this._getInputVal(), this._getFormatPattern()));
                ).bind(this)
```